modules/dataquery/ajax/saveQuery.php

```diff
@@ -20,8 +20,29 @@
 $client = new NDB_Client();
 $client->makeCommandLine();
 $client->initialize(__DIR__ . "/../../../project/config.xml");
-$user         = User::singleton();
+
+$user = User::singleton();
+$cdb  = CouchDB::singleton();
+$qid  = $user->getUserName() . "_" . $_REQUEST['QueryName'];
+
+if ($_REQUEST['SharedQuery'] === "true") {
+    $qid = "global:" . $qid;
+}
+
+if ($_REQUEST['OverwriteQuery'] === "false") {
+    $results = $cdb->getDoc(
+        $qid
```

Comment:
Could you link me to the documentation regarding escaping? I admit I don't know a lot about CouchDB.

However, one potential exploit would be to chain requests. A valid URI could be supplied and a new line character appended to issue a second request. 

Ultimately an input like "`validID\nGET /OurDB/sensitiveConfigInfo`" **might** resolve like the following:

> `GET /OurDB/validDocumentID\nGET /OurDB/sensitiveConfigInfo`

or 

> `GET /OurDB/validDocumentID`
>`GET /OurDB/sensitiveConfigInfo`

Of course this is just a high-level example and that syntax doesn't make sense. I don't know enough about the functioning or syntax of CouchDB to know if this is possible. 

However, someone out there might. I still think it's best practice to validate the input before passing it directly to a DB query.